nemo_curator/modules/semantic_dedup.py

```diff
@@ -199,21 +204,37 @@ def create_embeddings(
         return pipe(ddf)
 
     def __call__(self, dataset: DocumentDataset) -> DocumentDataset:
-        embedding_ddf = self.create_embeddings(dataset.df, self.input_column)
+        t0 = time.time()
         if self.write_embeddings_to_disk:
-            write_to_disk(
-                embedding_ddf,
-                self.embedding_output_dir,
-                write_to_filename=self.write_to_filename,
-                output_type="parquet",
-            )
-            return DocumentDataset(
+            with performance_report_if(
+                self.profile_dir,
+                f"embedding-creator-{datetime.now().strftime('%Y%m%d_%H%M%S')}.html",
+            ):
+                embedding_ddf = self.create_embeddings(dataset.df, self.input_column)
+                write_to_disk(
+                    embedding_ddf,
+                    self.embedding_output_dir,
+                    write_to_filename=self.write_to_filename,
+                    output_type="parquet",
+                )
+            dd = DocumentDataset(
```

Comment:
Minor nit: From  dask conventions it's very common to use something like `from dask import dataframe as dd` so I prefer not having it as a variable name and using something like `ddf` instead.